tests/Core/Infrastructure/SemanticVersionHelperTest.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/**
+ * Copyright © 2016-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+declare(strict_types=1);
+
+namespace CoreTest\Infrastructure;
+
+use Core\Infrastructure\SemanticVersionHelper;
+use PHPUnit\Framework\TestCase;
+
+class SemanticVersionHelperTest extends TestCase
+{
+    /**
+     * @dataProvider getMajorVersionProvider
+     *
+     * @param mixed $value
+     * @param ?int|int $expectedResult
+     *
+     * @return void
+     */
+    public function testGetMajorVersion($value, ?int $expectedResult): void
+    {
+        $this->assertSame($expectedResult, SemanticVersionHelper::getMajorVersion((string)$value));
+    }
+
+    /**
+     * @return array
+     */
+    public function getMajorVersionProvider(): array
+    {
+        return [
+            ['', null],
+            ['foo-bar-baz', null],
+            [555, null],
+            [-555, null],
+            ['1.0.0', 1],
+            ['1.0.2', 1],
+            ['1.2.2', 1],
+            ['5.2.222', 5],
+            ['55.28888.222', 55],
+            ['-55.28888.222', null],
+        ];
```

Comment:
Here is few more examples of the semantic versions: 

Valid Semantic Versions:
1.0.0
0.1.0
0.0.1
10.20.30
2.3.4-beta
5.6.7-alpha.123
1.2.3-rc.456+build789
3.2.1+build123
4.5.6-beta.789+build999

Invalid Semantic Versions:
1.2
1.2.a
1.2.3.4
1.2.3-rc!
1.2.3-beta@456
1.2.3+build_789
1.2.3-rc.456+build_789
1.2.3-beta+build!999
1.2.3-beta+build.999-456

Reply:
Thanks, I have added the cases to test 
but most of Invalid versions `composer/semver` determinate as valid